SamsungPay/src/main/java/com/braintreepayments/api/SamsungPayInternalClient.java

```diff
@@ -0,0 +1,164 @@
+        samsungPay.getSamsungPayStatus(new StatusListener() {
+            @Override
+            public void onSuccess(int statusCode, Bundle bundle) {
+                Exception samsungPayError = null;
+
+                @SamsungPayError int error = SamsungPayError.SAMSUNG_PAY_ERROR_UNKNOWN;
+                if (statusCode != SPAY_READY) {
+                    if (bundle != null && bundle.containsKey(SamsungPay.EXTRA_ERROR_REASON)) {
+                        int reason = bundle.getInt(SamsungPay.EXTRA_ERROR_REASON);
+                        switch (reason) {
+                            case ERROR_SPAY_APP_NEED_TO_UPDATE:
+                                error = SamsungPayError.SAMSUNG_PAY_APP_NEEDS_UPDATE;
+                                break;
+                            case ERROR_SPAY_SETUP_NOT_COMPLETED:
+                                error = SamsungPayError.SAMSUNG_PAY_SETUP_NOT_COMPLETED;
+                                break;
+                        }
+                    } else {
+                        if (statusCode == SPAY_NOT_READY) {
+                            error = SamsungPayError.SAMSUNG_PAY_NOT_READY;
+                        } else {
+                            // SPAY_NOT_SUPPORTED
+                            error = SamsungPayError.SAMSUNG_PAY_NOT_SUPPORTED;
+                        }
+                    }
+                    samsungPayError = new SamsungPayException(error);
+                }
+
+                callback.onResult(statusCode, samsungPayError);
+            }
+
+            @Override
+            public void onFail(int errorCode, Bundle bundle) {
+                SamsungPayException exception = new SamsungPayException(errorCode);
+                callback.onResult(null, exception);
+            }
+        });
+    }
+
+    void getAcceptedCardBrands(final GetAcceptedCardBrandsCallback callback) {
+        paymentManager.requestCardInfo(new Bundle(), new PaymentManager.CardInfoListener() {
+            @Override
+            public void onResult(final List<CardInfo> cardInfos) {
+                Set<SpaySdk.Brand> spayAcceptedCardBrands = new HashSet<>();
+                if (cardInfos != null) {
+                    for (CardInfo cardInfo : cardInfos) {
+                        spayAcceptedCardBrands.add(cardInfo.getBrand());
+                    }
+                }
+                // equivalent to getting the intersection of both sets
+                spayAcceptedCardBrands.retainAll(braintreeAcceptedCardBrands);
+                callback.onResult(new ArrayList<>(spayAcceptedCardBrands), null);
+            }
+
+            @Override
+            public void onFailure(int errorCode, Bundle bundle) {
+                Exception error = new SamsungPayException(errorCode);
+                callback.onResult(null, error);
+            }
+        });
+    }
+
+    void startSamsungPay(CustomSheetPaymentInfo customSheetPaymentInfo, final SamsungPayStartListener listener) {
```

Comment:
Would a more generic name like `SamsungPayListener` make sense here? It's a little misleading since the listener is also being called when the card info is updated, not just on start.

Reply:
I agree yes. We had to add a `Listener` style callback here because of the `cardInfoUpdated` method.

Reply:
Done!